actionview/lib/action_view/helpers/text_helper.rb

```diff
@@ -143,11 +143,13 @@ def highlight(text, phrases, options = {}, &block)
             Regexp === p ? p.to_s : Regexp.escape(p)
           end.join("|")
 
+          regex = /(?<!&)(#{match})(?![^<]*?>)/i
```

Comment:
The `(?<!&)` negative lookbehind as it is seems incomplete.

`highlight("x & y", "am")` would result in a desired `"x &amp; y"` output where nothing is highlighted. But `highlight("x & y", "mp")` would result in an undesired `"x &a<mark>mp</mark>; y"` highlighted output because `mp` is not immediately preceded by `&`. But it is still part of the overall `&amp;` html entity.